src/Microsoft.ML/Data/TextLoader.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Api;
+using Microsoft.ML.Runtime.Data;
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Text.RegularExpressions;
+
+namespace Microsoft.ML.Data
+{
+    public sealed partial class TextLoaderRange
+    {
+        public TextLoaderRange()
+        {
+        }
+
+        public TextLoaderRange(int ordinal)
```

Comment:
Please refer to \src\microsoft.ml.data\dataloadsave\text\textloader.cs line 149 - isValid method

---
In reply to: [188737039](https://github.com/dotnet/machinelearning/pull/142#discussion_r188737039) [](ancestors = 188737039)

Reply:
Hi Zeeshan! Just reactivating, sorry. I'm not certain I agree. When we have the sort of thing when we're building up through use of property setting, then sure, there's an expectation that we can't *really* validate much until `Pipeline.Train` is called, since of course literally anything could change up till that time, and it's a bit awkward to try to communicate what checks should go in property setters *anyway*. (E.g., I might imagine being able to test that `Min` and `Max` are above 0, but I couldn't necessarily check that `Min <= Max` in a setter, since I have no idea when that occurs.)

However in this case, we are talking about an actual constructor. We should fail as soon as we are able to *reasonably* detect that something is wrong. We can't do it there, but we can do it here.

---
In reply to: [188836637](https://github.com/dotnet/machinelearning/pull/142#discussion_r188836637) [](ancestors = 188836637,188737039)

Reply:
Just think about what a better experience is. Someone comes in, calls this constructor, builds a whole pipeline, does a bunch of stuff, does a lot of work, calls `Train` and *then* hits the error, or not.

---
In reply to: [189080344](https://github.com/dotnet/machinelearning/pull/142#discussion_r189080344) [](ancestors = 189080344,188836637,188737039)